src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -4622,6 +4622,19 @@ internal Token NextToken()
                         return ScanHereStringExpandable();
                     }
 
+                    if (ExperimentalFeature.IsEnabled("PSGeneralizedSplatting"))
+                    {
+                        if (c1 == '@')
+                        {
+                            var c2 = PeekChar();
+                            if (c2 == '{')
+                            {
+                                UngetChar();
+                                return NewToken(TokenKind.At);
```

Comment:
This is worth a short comment that it is putting back `@` (c1). Normally it looks wrong to see UngetChar shortly after PeekChar.

That said, the extra checks here seem unnecessary and getting in the way of fully generalized splatting. At a minimum, the code should have comments rationalizing why - because it seems to me we don't need any additional checks on `c1` here - we just return TokenKind.At straightaway.

Reply:
I've only added the additional lookahead for `@@{` just to be on the safe side. I've simplified it now as per your request to look ahead only for `@@` and added a comment about why we put the 2nd `@` token back.